Feature/#182 create edit profile page form logic

EspressoCode opened the pull request:
# Related Issue
- #182 create edit profile page form logic

# Proposed changes
- Added logic to edit profile component and the dialog in the home page, using the two backend endpoints for user photo change and the general update user, a user can edit profile info as well as the photo.

# Additional Info
- I made this pr for a high-level review of how the logic was implemented, so there will be a couple of changes and unit/e2e tests for the components which will be added in a future commit.

# Reviewer(s)
@Alessandro100

Alessandro100 commented on client/src/app/components/edit-profile/edit-profile.component.ts:
```diff
@@ -1,22 +1,234 @@
+  cancelPhoto(): void {
+    this.temporaryPhotoFile = null;
+    this.userPhotoURL = this.userInState.avatarURL;
+    console.log('remove photo clicked');
```
remove this console log

Alessandro100 reviewed (changes requested):
Very nicely implemented with NgRx, the code looks good. There are a couple of issues. For the picture, the user can choose any file type, we can force them to choose only image tags. Second thing is that the cover image functionality doesn't work

EspressoCode commented:
> Very nicely implemented with NgRx, the code looks good. There are a couple of issues. For the picture, the user can choose any file type, we can force them to choose only image tags. Second thing is that the cover image functionality doesn't work
I think I implemented that, so if you try it with a pdf file for example, you should see the error.
in the ts file:
`if (type !== 'image/jpeg' && type !== 'image/png') {
      this.pictureErrMessage =
        'The photo must be a file of type: jpeg, png, jpg!';
      this.userPhotoURL = this.userInState.avatarURL;
      this.temporaryPhotoFile = null;
      return;}`
in template file:
`<div class="profile-picture">
      <img [src]="userPhotoURL" alt="profile picture" class="profile-pic" />
      <button mat-icon-button class="edit-profile-pic" *ngIf="userPhotoURL==userInState.avatarURL">
        <input single type="file" id="profile-picture" style="display: none;" **accept="image/x-png,image/jpeg"**
          (change)="onFileSelected($event)" />
        <label for="profile-picture">
          <mat-icon>add_a_photo</mat-icon>
        </label>
      </button>`

EspressoCode commented:
In the commit before the last commit (3f61fde), I added cover photo to both the frontend and backend. This needs to be reviewed by the backend to make sure we can safely merge. Tests for backends also fixed based on new changes, so please review this pr and especially the last commit, thanks!
@razine-bensari @DPigeon @wltrf @abdulmansour

Alessandro100 reviewed (approved):
Nice job on the story! Good initiative on the backend section, and the tests look really good.
